File: python/python3/1961-isPrefixString.py
def isPrefixString(s: str, words: List[str]) -> bool:
    tmp = ''
    # python3 中使用 seperator.join(list)
    # 可以把列表转换成字符串(使用分隔符连接)
    sLen = len(tmp.join(words))
    wLen = len(words)
    for i in range(0, wLen):
        item = words[i]
        tmp = tmp + item;
        if tmp == s:
            return True
        if len(tmp) > sLen:
            return False
    return False

File: typescript/src/1961-isPrefixString.ts
function isPrefixString(s: string, words: string[]): boolean {
  const res: string = words.join('');
  if (res === s) return true;
  const sLen: number = s.length;
  if (res.length < sLen) {
    return false;
  }
  let tmp = '';
  let i: number;
  for (i = 0; i < words.length; i++) {
    const item: string = words[i];
    tmp = tmp + item;
    if (tmp === s) {
      return true;
    }
    if (tmp.length > sLen) {
      return false;
    }
  }
  return false;
}

export {isPrefixString};
